jaxdecomp/_src/padding.py

```diff
@@ -190,16 +189,11 @@ def per_shard_impl(arr: ArrayLike, padding_width: int | tuple[int]) -> Array:
     first_x, last_x = unpadding_width[0]
     first_y, last_y = unpadding_width[1]
     first_z, last_z = unpadding_width[2]
-    # Interior padding is padding between each row .. not needed here
-    interiour_padding = 0
-
-    # unlike jnp.pad lax.pad can unpad if given negative values
-    return lax.pad(
-        arr,
-        padding_value=0.0,
-        padding_config=((-first_x, -last_x, interiour_padding),
-                        (-first_y, -last_y, interiour_padding),
-                        (-first_z, -last_z, interiour_padding)))
+    last_x = arr.shape[0] - last_x
+    last_y = arr.shape[1] - last_y
+    last_z = arr.shape[2] - last_z
+
+    return arr[first_x:last_x, first_y:last_y, first_z:last_z]
```

Comment:
I guess this could be simplified to 
```suggestion
    return arr[first_x:-last_x, first_y:-last_y, first_z:-last_z]
```
but your suggestion is still clear

Reply:
Ok I remember why I did this
I had your version before and I had an issue when halo_size is 0 (no halo for a certain axis)
obviously I can have if branches to apply or not apply the unpadding, but I found that this way is clearer
do you agree?